docs/sources/flow/reference/components/prometheus.exporter.statsd.md

```diff
@@ -0,0 +1,108 @@
+
+## Usage
+
+```river
+prometheus.exporter.statsd "LABEL" {
+    
+}
+```
+
+## Arguments
+The following arguments can be used to configure the exporter's behavior.
+All arguments are optional. Omitted fields take their default values.
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`listen_udp`                                      | `string`       | The UDP address on which to receive statsd metric lines. Use "" to disable it. | `9125` | mo
+`listen_tcp`                                      | `string`       | The UDP address on which to receive statsd metric lines. Use "" to disable it. | `9125` | no
+`listen_unixgram`                                 | `string`       | The Unixgram socket path to receive statsd metric lines in datagram. Use "" to disable it. | | no
+`unix_socket_mode`                                | `string`       | The permission mode of the unix socket. | `755` | no
+`mapping_config_path`                             | `string`       | The path to a YAML mapping file used to translate specific dot-separated StatsD metrics into labeled Prometheus metrics. | | no
+`read_buffer`                                     | `int`          | Size (in bytes) of the operating system's transmit read buffer associated with the UDP or Unixgram connection. | | no
+`cache_size`                                      | `int`          | Maximum size of your metric mapping cache. Relies on least recently used replacement policy if max size is reached. | `1000` | no
+`cache_type`                                      | `string`       | Metric mapping cache type. Valid options are "lru" and "random". | `lru` | no
+`event_queue_size`                                | `int`          | Size of internal queue for processing events. | `10000` | no
+`event_flush_threshold`                           | `int`          | Number of events to hold in queue before flushing. | `1000`| no
+`event_flush_interval`                            | `string`       | Maximum time between event queue flushes. | `200ms`| no
+`parse_dogstatsd_tags`                            | `string`       | Parse DogStatsd style tags. Enabled by default. | `true`| no
+`parse_influxdb_tags`                             | `string`       | Parse InfluxDB style tags. Enabled by default. | `true`| no
+`parse_librato_tags`                              | `string`       | Parse Librato style tags. Enabled by default. | `true`| no
+`parse_signalfx_tags`                             | `string`       | Parse SignalFX style tags. Enabled by default. | `true`| no
+
+At least one of `listen_udp`, `listen_tcp` or `listen_unixgram` should be enabled.
+For details on how to use the mapping config file, please check the official [statsd_exporter docs](https://github.com/prometheus/statsd_exporter#metric-mapping-and-configuration).
+Please make sure the kernel parameter `net.core.rmem_max` is set to a value greater than the value specified in `read_buffer`.
+
```

Comment:
It hasn't been merged, but there's [a doc on writing flow component documentation](https://github.com/grafana/agent/pull/3145/files#diff-9739165705e77b11529c4aab59fbba6fb6ba5308f6b308ba1e6023cb222484aeR68) that indicates that a `Blocks` section should still exist for components that do not have any blocks.

[Here is an example](https://github.com/grafana/agent/pull/3145/files#diff-9739165705e77b11529c4aab59fbba6fb6ba5308f6b308ba1e6023cb222484aeR239) of what can be done in that case.